src/world_generation/rng_from_seed.rs

```diff
@@ -0,0 +1,21 @@
+use std::hash::{Hash, Hasher};
+
+use rand::{rngs::SmallRng, SeedableRng};
+
+
+pub struct RNGFromSeed {}
```

Comment:
The idea to put this in an "phantom struct" (*it's the name*) for encapsulation is good. Especially if there was several 'generators' : here we only have 1 generator, `rgn_seed_coord`

But to be fair I don't see why in the future we might add either (1) fields in the struct or (2)  add more generators.

So, let's keep it as simple as possible.

- Remove the struct `RNGFromSeed`
-  make a public fonction at the root of this file
- Specify in the name of the public method that it is a generator : `new_rng_seed_coord`